saphir/src/file/middleware.rs

```diff
@@ -0,0 +1,235 @@
+use crate::{
+    file::{
+        cache::FileCache,
+        conditional_request::{format_systemtime, is_fresh, is_precondition_failed},
+        etag::{EntityTag, SystemTimeExt},
+        range::Range,
+        range_requests::{extract_range, is_range_fresh, is_satisfiable_range},
+        Compression,
+    },
+    prelude::*,
+};
+use mime::Mime;
+use mime_guess::from_path;
+use percent_encoding::percent_decode;
+use std::{
+    path::{Path, PathBuf},
+    str::{FromStr, Utf8Error},
+    time::SystemTime,
+};
+
+const CACHE_MAX_FILE_SIZE: u64 = 2_147_483_648;
```

Comment:
Those are default, therefore they should be renamed as : 
DEFAULT_CACHE_MAX_FILE_SIZE & DEFAULT_ CACHE_MAX_CAPACITY

Also since those two are defaults I think that having a max file size of 2GB and a max Cache size of 8GB is pretty aggressive, I would go for something like 5MB per file & 512MB for the cache.